bootstrap/tracing.go

```diff
@@ -165,3 +241,22 @@ func Merge(maps ...map[string]any) map[string]any {
 
 	return merged
 }
+
+func (b *Bootstrap) toAttributes(extras map[string]any) []attribute.KeyValue {
+	attrs := make([]attribute.KeyValue, 0, len(extras))
+	for k, v := range extras {
+		switch v := v.(type) {
+		case string:
+			attrs = append(attrs, attribute.String(k, v))
+		case int:
+			attrs = append(attrs, attribute.Int(k, v))
+		case bool:
+			attrs = append(attrs, attribute.Bool(k, v))
+		default:
+			b.shell.Warningf("Unknown attribute type (key: %v, value: %v (%T)) passed when initialising OpenTelemetry. This is a bug, submit this error message at https://github.com/buildkite/agent/issues", k, v, v)
+			b.shell.Warningf("OpenTelemetry will still work as intended, but the attribute %v and its value above will not be included", v)
```

Comment:
Maybe not quite `as intended`?
```suggestion
			b.shell.Warningf("OpenTelemetry will still work, but the attribute %v and its value above will not be included", v)
```

Reply:
heh, yes, good point :P